### PR TITLE
Outgoing

### PR DESCRIPTION
**Pull request by MichaelSimkin**

### CODE REVIEW

**Comment by yarinBenisty** [comment]
done

File: src/express/outgoing/interface.ts [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 
 export interface IOutgoingFile {
     _id: mongoose.Types.ObjectId;
-    approvers: mongoose.Types.ObjectId[];
-    name: string;
-    fileId: mongoose.Types.ObjectId;
-    from: mongoose.Types.ObjectId;
-    to: mongoose.Types.ObjectId;
+    approvers: string[];
+    fileName: string;
```
Comment (MichaelSimkin) [inline src/express/outgoing/interface.ts:9]: Lets it to `name`
```suggestion
    name: string;
```

File: src/express/outgoing/manager.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * @returns {Promise<IOutgoingFile>} Promise object containing the created outgoing file.
+ */
+export const createOutgoingFile = (file: INewOutgoingFile): Promise<IOutgoingFile> => {
+    return outgoingFileModel.create(file);
+};
+
+/**
+ * Get an outgoing file.
+ * @param fileId - The file id.
+ * @returns {Promise<IOutgoingFile>} - Promise containing the outgoing file.
```
Comment (MichaelSimkin) [inline src/express/outgoing/manager.ts:19]:
```suggestion
 * @returns {Promise<IOutgoingFile>} Promise containing the outgoing file.
```

File: src/express/outgoing/manager.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+export const getOutgoingFileById = async (fileId: mongoose.Types.ObjectId): Promise<IOutgoingFile> => {
+    const result = await outgoingFileModel.findById(fileId).exec();
+    if (!result) throw new ServerError(StatusCodes.NOT_FOUND, 'File not found');
+    return result;
+};
+
+/**
+ * Get outgoing files.
+ * @param query - Outgoing file filters.
+ * @returns {Promise<IOutgoingFile[]>} - Promise containing the outgoing files arrays.
```
Comment (MichaelSimkin) [inline src/express/outgoing/manager.ts:30]:
```suggestion
 * @returns {Promise<IOutgoingFile[]>} Promise containing the outgoing files arrays.
```

File: src/express/outgoing/manager.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ */
+export const getOutgoingFiles = async (query: IOutgoingFileFilters): Promise<IOutgoingFile[]> => {
+    return outgoingFileModel.find(query).exec();
+};
+
+/**
+ * Delete an outgoing file.
+ * @param fileId - The file id.
+ * @returns {Promise<IOutgoingFile>} - Promise object containing the deleted file.
```
Comment (MichaelSimkin) [inline src/express/outgoing/manager.ts:39]:
```suggestion
 * @returns {Promise<IOutgoingFile>} Promise object containing the deleted file.
```

File: src/express/outgoing/model.ts [hunk 5]
```diff
@@ -6,39 +6,43 @@ import { IOutgoingFile } from './interface';
 const outgoingFileSchema = new mongoose.Schema<IOutgoingFile & mongoose.Document>(
     {
         approvers: {
-            objectId: { type: 'ObjectId', required: true },
+            type: [String],
+            required: true,
         },
-        name: {
+        fileName: {
```
Comment (MichaelSimkin) [inline src/express/outgoing/model.ts:12]: lets change it to name
```suggestion
        name: {
```

**Review by MichaelSimkin — COMMENTED** [review]
Excellent, just some small final nitpicks.

File: src/express/outgoing/interface.ts (hunk 1 above)
Comment (yarinBenisty) [reply src/express/outgoing/interface.ts:9]: done

File: src/express/outgoing/manager.ts (hunk 2 above)
Comment (yarinBenisty) [reply src/express/outgoing/manager.ts:19]: done

File: src/express/outgoing/manager.ts (hunk 3 above)
Comment (yarinBenisty) [reply src/express/outgoing/manager.ts:30]: done

File: src/express/outgoing/manager.ts (hunk 4 above)
Comment (yarinBenisty) [reply src/express/outgoing/manager.ts:39]: done

File: src/express/outgoing/model.ts (hunk 5 above)
Comment (yarinBenisty) [reply src/express/outgoing/model.ts:12]: done

**Pull request closed without merging**